src/Search/Workflow/JsonSerializeTransport.php

```diff
@@ -11,15 +11,28 @@ trait JsonSerializeTransport
 
     abstract public function getSdkClass() : string;
 
-    public function deserialize(string $json)
+    private function checkSerializer()
     {
-        if (
-            !isset($this->serializer) ||
-            null === $this->serializer ||
-            !$this->serializer instanceof Serializer
-        ) {
-            throw new LogicException('You must inject API SDK serializer for this to work (property: $serializer missing.)');
+        static $checked = false;
+
+        if (!$checked) {
+            if (
+                !isset($this->serializer) ||
+                null === $this->serializer ||
+                !$this->serializer instanceof Serializer
+            ) {
+                throw new LogicException('You must inject API SDK serializer for this to work (property: $serializer missing.)');
+            }
+            $checked = true;
         }
+
+        return $checked;
```

Comment:
Thought this was a better way of performing the check just once.

Reply:
Seems unnecessary to me. The checks being done are cheap, probably taking less than a single ms. I would recommend reverting it.

Reply:
It wasn't really about saving time but removing redundancy. Will reinstate, keeping it in it's own function but running on each call.